### DISCUSSION TITLE
NimBLE

### DISCUSSION THREAD

**Discussion by er-bsman** [Arulprasath36/WhatsUp_Development#674] [Ideas]
@h2zero,
hi,I have got the services, characteristics when scanned the ble device using the Mobile app.

Now, I used the NimBLEClient::discoverAttributes() fun to discover the service of the device. I want to get the Characteristics and want to use the GATT wite() fun. Kindly help me out, how to do it.

**Comment by h2zero** [comment]
Once you have a client connected you can write to a characteristic by calling `setValue` from there or get a reference to the service and then the characteristic and calling write on it. 

Have a look at the examples and documentation to see how this is done.

**Comment by er-bsman** [comment]
hi,i did tried. I got the services and could not get the characteristics, descriptor using the services. 

Is it something like this i need to do?
![Screenshot 2022-10-27 210532](https://user-images.githubusercontent.com/110908063/198377327-c1465dcf-7e3c-4fb1-aed8-0ad62d64cb55.png)

**Reply by h2zero** [reply]
Yes that's one way to do it. Give it a try and see what happens, let me know if you have an issue.

**Reply by er-bsman** [reply]
I tried to get the characteristics from the services but I could not. I would appreciate, if can you provide with me some examples or documentation on how to do it. My only concern is:

As I mentioned earlier, I have got the services, and I need to get the characteristics of that by the reference of the services and then to the descriptor after that I need to GATT write  to be done.

**Reply by er-bsman** [reply]
I managed to get it all the characteristics and the descriptor related to the services. Thanks.
